apps/plotting.py

```diff
@@ -750,178 +780,184 @@ def draw_lightcurve(switch: int, pathname: str, object_data, object_upper, objec
                 apparent_flux(*args) for args in zip(
                     mag.astype(float).values,
                     err.astype(float).values,
-                    pdf['i:magnr'].astype(float).values,
+                    pdf['i:magnr'].astype(float).values if is_dc_corrected else 99.0,
                     pdf['i:sigmagnr'].astype(float).values,
                     pdf['i:isdiffpos'].values
                 )
             ]
         )
+
         layout_lightcurve['yaxis']['title'] = 'Apparent DC flux (milliJansky)'
         layout_lightcurve['yaxis']['autorange'] = True
         scale = 1e3
 
-    hovertemplate = r"""
-    <b>%{yaxis.title.text}</b>: %{customdata[1]}%{y:.2f} &plusmn; %{error_y.array:.2f}<br>
-    <b>%{xaxis.title.text}</b>: %{x|%Y/%m/%d %H:%M:%S.%L}<br>
-    <b>mjd</b>: %{customdata[0]}
-    <extra></extra>
-    """
+    layout_lightcurve['shapes'] = []
+
     figure = {
-        'data': [
+        'data': [],
+        "layout": layout_lightcurve
+    }
+
+    for fid,fname,color in (
+            (1, "g", COLORS_ZTF[0]),
+            (2, "r", COLORS_ZTF[1])
+    ):
+        # High-quality measurements
+        hovertemplate = r"""
+        <b>%{yaxis.title.text}</b>: %{customdata[1]}%{y:.2f} &plusmn; %{error_y.array:.2f}<br>
+        <b>%{xaxis.title.text}</b>: %{x|%Y/%m/%d %H:%M:%S.%L}<br>
+        <b>mjd</b>: %{customdata[0]}
+        <extra></extra>
+        """
+        idx = pdf['i:fid'] == fid
+        figure['data'].append(
             {
-                'x': dates[pdf['i:fid'] == 1],
-                'y': mag[pdf['i:fid'] == 1] * scale,
+                'x': dates[idx],
+                'y': mag[idx] * scale,
                 'error_y': {
                     'type': 'data',
-                    'array': err[pdf['i:fid'] == 1] * scale,
+                    'array': err[idx] * scale,
                     'visible': True,
-                    'color': COLORS_ZTF[0]
+                    'width': 0,
+                    'opacity': 0.5,
+                    'color': color
                 },
                 'mode': 'markers',
-                'name': 'g band',
+                'name': '{} band'.format(fname),
                 'customdata': np.stack(
                     (
-                        pdf['i:jd'].apply(lambda x: x - 2400000.5)[pdf['i:fid'] == 1],
-                        pdf['i:isdiffpos'].apply(lambda x: '(-) ' if x == 'f' else '')[pdf['i:fid'] == 1],
+                        pdf['i:jd'].apply(lambda x: x - 2400000.5)[idx],
+                        pdf['i:isdiffpos'].apply(lambda x: '(-) ' if x == 'f' else '')[idx],
                     ), axis=-1
                 ),
                 'hovertemplate': hovertemplate,
-                "legendgroup": "g band",
-                'legendrank': 110,
+                "legendgroup": "{} band".format(fname),
+                'legendrank': 100 + 10*fid,
```

Comment:
It is probably not needed now that I re-arranged the code to include data points in proper order - but in general yes, it may be adjusted like that for arbitrary order of insertions of new traces